### PR TITLE
DB/Quest: Heroes of Old

### PR DESCRIPTION
**Pull request by poszer**
- Fix for the quest Heroes of Old

Co-Authored-By: Knindza <knindzagxg@users.noreply.github.com>

Quest Heroes of Old Fix 

##### CHANGES PROPOSED:

-  SAI update
-  

###### ISSUES ADDRESSED:

Closes https://github.com/azerothcore/azerothcore-wotlk/issues/1514

##### TESTS PERFORMED:

Yes. Tested in game, everything work. OS : Win
[Video here](https://www.youtube.com/watch?v=5NCzJSiE8Xs&feature=youtu.be)

##### HOW TO TEST THE CHANGES:

1. .go creature 35237
2. .quest add 2681
3. .quest complete 2681 and reward it
4. Take the quest "Hero of old"  now, speak with the fallen hero of the horde and click " Hero, i have a urgent business with Thund Splithoof" ( The Npc should be spawned )

##### KNOWN ISSUES AND TODO LIST:

- [ ]
- [ ] 

##### Target branch(es):

Master

Thanks @Knindzagxg

### CODE REVIEW

**Comment by poszer** [comment]
> Fall from Grace and A Tale of Sorrow can't be completed
> 
> Fix: (added comments too)
> 
> ```sql
> DELETE FROM `smart_scripts` WHERE `entryorguid`=7572;
> INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `event_param5`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_param4`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES (7572, 0, 0, 2, 62, 0, 100, 1, 842, 0, 0, 0, 0, 26, 2784, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 0, 'Fallen Hero of the Horde - On Gossip Option 0 Selected - Quest Credit \'Fall From Grace\''),
> (7572, 0, 1, 3, 62, 0, 100, 1, 881, 0, 0, 0, 0, 26, 2801, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 0, 'Fallen Hero of the Horde - On Gossip Option 1 Selected - Quest Credit \'A Tale of Sorrow\''),
> (7572, 0, 2, 0, 61, 0, 100, 0, 0, 0, 0, 0, 0, 72, 0, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 0, 'Fallen Hero of the Horde - On Gossip Option 0 Selected - Close Gossip'),
> (7572, 0, 3, 0, 61, 0, 100, 0, 0, 0, 0, 0, 0, 72, 0, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 0, 'Fallen Hero of the Horde - On Gossip Option 1 Selected - Close Gossip'),
> (7572, 0, 4, 5, 62, 0, 100, 0, 840, 2, 0, 0, 0, 11, 11024, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 'Fallen Hero of the Horde - On Gossip Option Selected - Cast Self Call of Thund'),
> (7572, 0, 5, 0, 61, 0, 100, 0, 0, 0, 0, 0, 0, 72, 0, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 0, 'Fallen Hero of the Horde - On Gossip Option Selected - Close Gossip');
> ```

Thanks @wizzymore

**Comment by Knindzagxg** [comment]
```sql
UPDATE `creature_template` SET `AIName` = 'SmartAI' WHERE `entry` = 7750;

DELETE FROM `smart_scripts` WHERE (source_type = 0 AND entryorguid = 7750);
INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
(7750, 0, 0, 0, 20, 0, 100, 0, 2701, 0, 0, 0, 41, 5000, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, ''),
(7750, 0, 1, 2, 19, 0, 100, 0, 2701, 0, 0, 0, 70, 0, 0, 0, 0, 0, 0, 14, 44733, 141980, 0, 0, 0, 0, 0, ''),
(7750, 0, 2, 0, 61, 0, 100, 0, 0, 0, 0, 0, 70, 0, 0, 0, 0, 0, 0, 14, 44732, 141981, 0, 0, 0, 0, 0, '');```

To remove Tauren once u complete quest.

**Comment by poszer** [comment]
> ```sql
> UPDATE `creature_template` SET `AIName` = 'SmartAI' WHERE `entry` = 7750;
> 
> DELETE FROM `smart_scripts` WHERE (source_type = 0 AND entryorguid = 7750);
> INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
> (7750, 0, 0, 0, 20, 0, 100, 0, 2701, 0, 0, 0, 41, 3000, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, ''),
> (7750, 0, 1, 2, 19, 0, 100, 0, 2701, 0, 0, 0, 70, 0, 0, 0, 0, 0, 0, 14, 44733, 141980, 0, 0, 0, 0, 0, ''),
> (7750, 0, 2, 0, 61, 0, 100, 0, 0, 0, 0, 0, 70, 0, 0, 0, 0, 0, 0, 14, 44732, 141981, 0, 0, 0, 0, 0, '');```
> 
> To remove Tauren once u complete quest.
> ```

Updated, thanks

File: data/sql/updates/pending_db_world/rev_1553294024713457200.sql [hunk 1]
```diff
@@ -6,4 +6,4 @@ DELETE FROM `smart_scripts` WHERE (source_type = 0 AND entryorguid = 7750);
 INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
 (7750, 0, 0, 0, 20, 0, 100, 0, 2701, 0, 0, 0, 41, 3000, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, ''),
 (7750, 0, 1, 2, 19, 0, 100, 0, 2701, 0, 0, 0, 70, 0, 0, 0, 0, 0, 0, 14, 44733, 141980, 0, 0, 0, 0, 0, ''),
-(7750, 0, 2, 0, 61, 0, 100, 0, 0, 0, 0, 0, 70, 0, 0, 0, 0, 0, 0, 14, 44732, 141981, 0, 0, 0, 0, 0, '');```
```
Comment (FrancescoBorzi) [inline data/sql/updates/pending_db_world/rev_1553294024713457200.sql]: @Knindzagxg those are only supposed to be used on github comments, not in real code :)

**Comment by poszer** [comment]
Same like for previous PR. This is already tested by @Knindzagxg , @wizzymore  and me. Can be merged

**Comment by comix1988** [comment]
@poszer i tested in a master branch but not work. (this fix is already merged)

https://youtu.be/4mz4pDIUEsI

**Comment by poszer** [comment]
> @poszer i tested in a master branch but not work. (this fix is already merged)
> 
> https://youtu.be/4mz4pDIUEsI

Technically, it's four chain quests that tie into each other. Do the previous quests first and try than :+1:

**Comment by wizzymore** [comment]
@comix1988 
do this or it won't work
.go creature 35237
.quest add 2681
.quest complete 2681 and reward it
Take the quest "Hero of old" now, speak with the fallen hero of the horde and click " Hero, i have a urgent business with Thund Splithoof" ( The Npc should be spawned )

**Comment by poszer** [comment]
@comix1988  Also, take a look on video, just recorded :
[Heroes of Old Questchain - AzerothCore
](https://youtu.be/cX-_ZvSMcuk)

**Comment by comix1988** [comment]
@wizzymore I tried to do what you wrote to me but it doesn't work.
I'm testing on ubuntu.

**Pull request merged**